server/mailrepository/mailrepository-api/src/main/java/org/apache/james/mailrepository/api/MailRepositoryUrl.java

```diff
@@ -38,22 +41,41 @@ public static final MailRepositoryUrl from(String url) {
         return new MailRepositoryUrl(url);
     }
 
+    public static MailRepositoryUrl fromPathAndProtocol(MailRepositoryPath path, String protocol) {
+        return new MailRepositoryUrl(path, protocol);
+    }
+
     private final String value;
+    private final MailRepositoryPath path;
     private final Protocol protocol;
 
     private MailRepositoryUrl(String value) {
         Preconditions.checkNotNull(value);
-        Preconditions.checkArgument(value.contains(":"), "':' is mandatory to delimit protocol");
+        Preconditions.checkArgument(value.contains(PROTOCOL_SEPARATOR), "The expected format is: <protocol> \"" + PROTOCOL_SEPARATOR + "\" <path>");
         this.value = value;
-        this.protocol = new Protocol(Splitter.on(':')
+        String protocol = Splitter.on(':')
```

Comment:
why don't you split on PROTOCOL_SEPARATOR?

Reply:
So true, thanks.